internal/operators/ocs/validations.go

```diff
@@ -156,16 +169,19 @@ func gbToBytes(gb int64) int64 {
 }
 
 func (o *ocsValidator) validateOCS(log logrus.FieldLogger, hosts []*models.Host, cpu int64, RAM int64, disk int64, hostsWithDisks int64) (bool, string) {
-
+	var TotalCPUs int64
+	var TotalRAM int64
 	var status string
 	if int64(len(hosts)) == o.OCSRequiredHosts { // for 3 hosts
-		if cpu < o.OCSRequiredCompactModeCPUCount || RAM < gbToBytes(o.OCSRequiredCompactModeRAMGB) || disk < o.OCSRequiredDisk || hostsWithDisks < o.OCSRequiredHosts { // check for master nodes requirements
+		TotalCPUs = o.OCSRequiredCompactModeCPUCount + disk*o.OCSRequiredDiskCPUCount                                        // need additional 2 cpus per disk
+		TotalRAM = o.OCSRequiredCompactModeRAMGB + disk*o.OCSRequiredDiskRAMGB                                               // need additional 5 GB per disk
+		if cpu < TotalCPUs || RAM < gbToBytes(TotalRAM) || disk < o.OCSRequiredDisk || hostsWithDisks < o.OCSRequiredHosts { // check for master nodes requirements
```

Comment:
This validation doesn't seem right to me - we should be checking that each host has the required resources for its disks, not looking at totals for the cluster (for example, we can have one host with many disks and little CPU, and another with few disks and lots of CPU).

Reply:
fixed